extensions/positron-assistant/src/utils.ts

```diff
@@ -89,17 +92,34 @@ export function toAIMessage(
 						if (toolCall.name === PositronAssistantToolName.GetPlot) {
 							aiMessages.push(getPlotToolResultToAiMessage(part));
 						} else {
-							aiMessages.push({
-								role: 'tool',
+							// Convert part.content (an array of LanguageModelParts) to a string for non-experimental providers
+							// AI SDK 5 expects output in structured format: { type: 'text' | 'json', value: ... }
+							const textContent = part.content
+								.filter((c): c is vscode.LanguageModelTextPart => c instanceof vscode.LanguageModelTextPart)
+								.map(c => c.value)
+								.join('');
+
+							// Try to parse as JSON if it looks like JSON, otherwise use as text
+							let output: { type: 'json'; value: unknown } | { type: 'text'; value: string };
```

Comment:
There is a cast down below which can be removed if this type is changed:

```suggestion
							let output: { type: 'json'; value: ai.JSONValue } | { type: 'text'; value: string };
```